data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/BinaryOperator.java

```diff
@@ -0,0 +1,44 @@
+package org.opensearch.dataprepper.expression;
+
+import org.antlr.v4.runtime.RuleContext;
+import org.opensearch.dataprepper.expression.antlr.DataPrepperExpressionParser;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+abstract class BinaryOperator<T> implements Operator<T> {
```

Comment:
Do we need an abstract BinaryOperator now that the GenericNegateOperator no longer extends this class? The only class extending this class is the `GenericEqualOperator`. All of this logic can just be pulled in and we can eliminate a layer of abstraction and simplify the code.

Reply:
After updating the GenericNotEqualOperator, the `BinaryOperator` class is unnecessary. We could remove it in a future PR or update other binary operators to extend it. The `BinaryOperator` class can remove a significant amount of repeated code. Lets choose a direction and I'll open a github issue.